smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/config/DefaultClientConfiguration.kt

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+package software.amazon.smithy.swift.codegen.config
+
+import software.amazon.smithy.codegen.core.Symbol
+import software.amazon.smithy.swift.codegen.ClientRuntimeTypes
+import software.amazon.smithy.swift.codegen.SwiftDependency
+import software.amazon.smithy.swift.codegen.SwiftTypes
+import software.amazon.smithy.swift.codegen.config.ClientConfiguration.Companion.runtimeSymbol
+import software.amazon.smithy.swift.codegen.model.toNullable
+
+class DefaultClientConfiguration : ClientConfiguration {
+    override val swiftProtocolName: Symbol
+        get() = runtimeSymbol("DefaultClientConfiguration", SwiftDependency.CLIENT_RUNTIME)
+
+    override val properties: Set<ConfigProperty>
+        get() = setOf(
+            ConfigProperty("encoder", ClientRuntimeTypes.Serde.RequestEncoder.toNullable()),
+            ConfigProperty("decoder", ClientRuntimeTypes.Serde.ResponseDecoder.toNullable()),
```

Comment:
RequestEncoder / ResponseDecoder are no longer used & no longer supplied to clients when the wire protocol for requests / responses is XML.

(Eventually the plan is to eliminate them entirely.)

Reply:
In any case, even where they are used, they do not need to be publicly exposed; they're an implementation detail of our wire protocols and don't need to be exposed for modularization or any other interoperability-related purpose.